internal/core/server/kernel.go

```diff
@@ -28,6 +30,45 @@ type Kernel struct {
 	TaskTracker *task.TaskTracker
 }
 
+func (kern *Kernel) DescribeTemplates(ctx context.Context, input *api.DescribeTemplatesInput) (*api.DescribeTemplatesOutput, error) {
+	return &api.DescribeTemplatesOutput{Templates: template.GetTemplateDescriptions()}, nil
+}
+
+func (kern *Kernel) CreateProject(ctx context.Context, input *api.CreateProjectInput) (*api.CreateProjectOutput, error) {
+	projectDir := input.Root
+	if !filepath.IsAbs(projectDir) {
+		return &api.CreateProjectOutput{}, errors.New("path must be absolute")
+	}
+
+	var templateDir string
+	if input.TemplateUrl != nil {
+		var err error
+		templateDir, err = template.GetTemplateFiles(ctx, *input.TemplateUrl)
+		if err != nil {
+			return &api.CreateProjectOutput{}, fmt.Errorf("getting template files: %w", err)
+		}
+	}
+
+	err := os.Mkdir(projectDir, 0750)
+	if err != nil {
+		return &api.CreateProjectOutput{}, fmt.Errorf("making project dir: %w", err)
+	}
+
+	if templateDir != "" {
+		copy.Copy(templateDir, projectDir, copy.Options{
+			OnSymlink: func(string) copy.SymlinkAction {
+				return copy.Deep
```

Comment:
Why Deep? Seems like `copy.Shallow` is more appropriate.

Reply:
No longer relevant since I changed this to a move — but you're probably right.